src/Roslyn.Diagnostics.Analyzers/Core/DeclarePublicAPIAnalyzer.Impl.cs

```diff
@@ -359,6 +390,49 @@ internal void OnCompilationEnd(CompilationAnalysisContext context)
                 }
             }
 
+            private void ProcessTypeForwardedAttributes(Compilation compilation, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken)
+            {
+                var typeForwardedToAttribute = compilation.GetTypeByMetadataName("System.Runtime.CompilerServices.TypeForwardedToAttribute");
+
+                if (typeForwardedToAttribute != null)
+                {
+                    foreach (var attribute in compilation.Assembly.GetAttributes())
+                    {
+                        if (typeForwardedToAttribute.Equals(attribute.AttributeClass))
+                        {
+                            var forwardedType = attribute.ConstructorArguments.Single().Value as INamedTypeSymbol;
```

Comment:
:question: Will this work correctly if the forwarded type is later forwarded again?